### PR TITLE
update 5 files

### PR DESCRIPTION
**Pull request by ghost**
Particle.py, params.py, particle_simulation.py, run_md_simulation_3, mpihostsfile

### CODE REVIEW

**Comment by nickfong** [comment]
I'll do a more thorough review of this in the next 96 hours or so, but
I'd suggest updating this patch to remove all of the testing/deprecated
code that is commented out.


I'd also suggest rebasing your fork's branch off of this upstream to
avoid the extraneous merge commit.


Best,

Nick



On Mon, 13 Mar 2017 17:44 -0700, boredHuman <notifications@github.com> wrote:
> Particle.py, params.py, particle_simulation.py, run_md_simulation_3,
> mpihostsfile
> 

> You can view, comment on, or merge this pull request online at:



>   https://github.com/NERSC/CompactCori/pull/8



> Commit Summary


>  * Merge pull request #1 from NERSC/master
>  * update Particle.py, params.py, particle_simulation.py,
>    run_md_simulation_3, mpihostsfile to the simulation
> File Changes


>  * *M* app/particle/Particle.py[1] (3)
>  * *M* app/particle/params.py[2] (1)
>  * *M* app/particle/particle_simulation.py[3] (214)
>  * *A* app/particle/run_md_simulation_3[4] (61)
>  * *A* mpihostsfile[5] (18)
> Patch Links:


>  * https://github.com/NERSC/CompactCori/pull/8.patch
>  * https://github.com/NERSC/CompactCori/pull/8.diff
> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub[6], or mute the
> thread[7].
> 




Links:

  1. https://github.com/NERSC/CompactCori/pull/8/files#diff-0
  2. https://github.com/NERSC/CompactCori/pull/8/files#diff-1
  3. https://github.com/NERSC/CompactCori/pull/8/files#diff-2
  4. https://github.com/NERSC/CompactCori/pull/8/files#diff-3
  5. https://github.com/NERSC/CompactCori/pull/8/files#diff-4
  6. https://github.com/NERSC/CompactCori/pull/8

File: app/particle/Particle.py [hunk 1]
```diff
@@ -82,7 +83,7 @@ def update_velocity(self, particles):
                 if euclidean_distance == 0:
                     force = max_force
                 else:
-                    force = params.force * (1 * (distances[i])/euclidean_distance**3)/self.mass
+                    force = params.force * distances[i] / euclidean_distance**3 / self.mass
```
Comment (nickfong) [inline app/particle/Particle.py]: While it doesn't make a difference at runtime, adding parenthesis here would make the code more readable.  

Also, Python groups left to right when handling precedence for multiplication and division; dividing something that is the quotient of a product is equivalent to multiplying the quotient by the aforementioned product.

File: app/particle/particle_simulation.py [hunk 2]
```diff
@@ -78,58 +79,94 @@
 params.init_total_energy = 0.0
 params.curr_total_energy = 0.0
 
+def add_one_particle(particle_id):
+    radius = 30#random.randint(1, params.max_radius)
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: The commented code here is extraneous

File: app/particle/particle_simulation.py [hunk 3]
```diff
@@ -78,58 +79,94 @@
...
+    position = [random.randint(radius, params.simulation_width - 1),
+                random.randint(radius, params.simulation_height - 1),
+                random.randint(radius, params.simulation_depth - 1)]
+    velocity = [400*random.randint(0,radius//4),
+                400*random.randint(0,radius//4),
+                400*random.randint(0,radius//4)]
+    mass = 3#random.randint(1,10)
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: As mentioned above, the commented code here is extraneous

File: app/particle/particle_simulation.py [hunk 4]
```diff
@@ -78,58 +79,94 @@
...
+def subtract_one_particle(particle_id):
+    if params.rank is 0:
+        for i in range(1, params.num_active_workers+1):
+            new_particles = params.comm.recv(source = mpi.ANY_SOURCE, status = params.mpi_status, tag = 0)
+            params.partitions[params.mpi_status.Get_source()].particles = new_particles
+        params.init_total_energy -= params.comm.recv(source = mpi.ANY_SOURCE, status = params.mpi_status, tag = 88)
+    elif params.rank <= params.num_active_workers:
+        partition = params.partitions[params.rank]
+        particles = partition.particles
+#        particle_to_be_removed = None
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Code that is commented out can be removed; everything should be in version control

File: app/particle/particle_simulation.py [hunk 5]
```diff
@@ -78,58 +79,94 @@
...
+        for particle in particles:
+            if particle.particle_id == particle_id:
+                params.comm.send(particle.particle_initial_energy, tag = 88)
+                particles.remove(particle)
+#                particle_to_be_removed = particle
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Ditto

File: app/particle/particle_simulation.py [hunk 6]
```diff
@@ -78,58 +79,94 @@
...
+                break
+#            else:
+#                util.info('particle_id ' + str(particle.particle_id) + ' is not in partition ' + str(params.rank))
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Ditto

File: app/particle/particle_simulation.py [hunk 7]
```diff
@@ -78,58 +79,94 @@
...
-        thread_num = util.determine_particle_thread_num(position[0])
-        new_particle = Particle(i, thread_num, position, velocity, mass, radius)
-        params.partitions[thread_num].add_particle(new_particle)
+        add_one_particle(i)
 
     params.curr_total_energy = params.init_total_energy
 
-def update_params():
+def update_new_num_particles():
+#    util.debug("update_new_num_particles: starting")
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Debug statements can be removed as well

File: app/particle/particle_simulation.py [hunk 8]
```diff
@@ -78,58 +79,94 @@
...
+    params.new_num_particles = params.comm.bcast(params.new_num_particles)
+#    util.debug("update_new_num_particles: ending")
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Ditto

File: app/particle/particle_simulation.py [hunk 9]
```diff
@@ -78,58 +79,94 @@
...
+
+def update_new_num_active_workers():
+#    util.debug("update_new_num_active_workers: starting")
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Ditto

File: app/particle/particle_simulation.py [hunk 10]
```diff
@@ -168,13 +205,32 @@ def change_num_active_workers():
...
-        print("receiving "+str(params.rank))
+#        print("receiving "+str(params.rank))
         params.partitions[params.rank].receive_new_particles()
-        #print("received "+str(params.rank))
+#        print("received "+str(params.rank))
+
+def change_num_particles():
+#    util.debug("change_num_particles is called")
+    if params.num_particles == params.new_num_particles:
+        pass
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: This is dead code

File: app/particle/particle_simulation.py [hunk 11]
```diff
@@ -168,13 +205,32 @@ def change_num_active_workers():
...
+        if params.rank is 0:
+            for i in range(params.num_particles, params.new_num_particles):
+                add_one_particle(i)
+            for i in range(1, params.num_threads):
+#                print("sending " + str(i))
+                params.comm.send(params.partitions[i].particles, dest = i, tag = 11)
+        else:
+            params.partitions[params.rank].receive_new_particles()
+    else:
+        # this is the case when params.num_particles > params.new_num_particles
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: If this is the case (where the antecedent of "this" is)
> this is the case when params.num_particles > params.new_num_particles

you can break up the two cases into `params.num_particles` is strictly greater than `params.new_num_particles` and `params.num_particles` is strictly less than `params.new_num_particles`

File: app/particle/particle_simulation.py [hunk 12]
```diff
@@ -168,13 +205,32 @@ def change_num_active_workers():
...
+        # note that params.new_num_particles == params.num_particles - 1
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Is this always guaranteed to be the case?  i.e. is only one particle allowed to be removed at a time?

File: app/particle/particle_simulation.py [hunk 13]
```diff
@@ -189,8 +245,8 @@ def do_GET(self):
             self.send_header("Access-Control-Allow-Origin", "*")
             self.end_headers()
             self.wfile.write(message.encode("utf-8"))
-        else:
-            util.info("GET sent to " + str(parsed_path[2]))
+#        else:
+#            util.info("GET sent to " + str(parsed_path[2]))
```
Comment (nickfong) [inline app/particle/particle_simulation.py]: Code that is commented out can be removed; everything should be in version control should the desire to revert changes arise.

**Review by nickfong — CHANGES_REQUESTED** [review]
A couple minor stylistic suggestions, and a couple places where some refactoring could be done, but nothing major.  Toward the end I stopped noting code that was commented out and should be removed before being checked in to the `master` branch of this upstream repo.  

I'm not sure how you went from having 2 commits to 6 (it seems that they are all duplicates of one another); if you'd like we can schedule some time in the next couple days to do a call where I can help you clean up your fork's history so we don't end up with extraneous (and duplicate) commits in this upstream.

**Comment by ghost** [comment]
i am not sure what i did, but i will try to resolve it

**Comment by ghost** [comment]
do you know how i can send private messages to you on github? i am unable to find out how =(

**Comment by nickfong** [comment]
Send me an email at the email address I left in my last email to
everyone.  There unfortunately aren't direct messages on GH.


On Tue, 21 Mar 2017 12:38 -0700, boredHuman <notifications@github.com> wrote:
> do you know how i can send private messages to you on github? i am
> unable to find out how =(
> — You are receiving this because your review was requested. Reply to
> this email directly, view it on GitHub[1], or mute the thread[2].
> 




Links:

  1. https://github.com/NERSC/CompactCori/pull/8#issuecomment-288194592

**Pull request closed without merging**